pokemongo_bot/cell_workers/incubate_eggs.py

```diff
@@ -57,6 +60,17 @@ def _apply_incubators(self):
             for egg in self.eggs:
                 if egg["used"] or egg["km"] == -1:
                     continue
+                
+                if self.breakable_incubator:
+                    if incubator.get('uses_remaining') is not None:
+                        if egg["km"] not in self.breakable_incubator:
+                            continue
+                    
+                if self.infinite_incubator:
+                    if incubator.get('uses_remaining') is None: 
```

Comment:
If you use the option breakable
   if the incubator is of type breakable
      if the number of km is not in the list breakable
         then stop and pass to the next egg

Same for infinite.

That's the key logic of the pr, but maybe I'm mistaken somewhere? What did you meant precisely?